### PR TITLE
TextInput - Display cursor when text input is in editing mode

### PR DESCRIPTION
**Pull request by martunjai**
Updated code to Display cursor when text input is in editing mode only.

### CODE REVIEW

File: ReactSkia/components/RSkComponentTextInput.cpp [hunk 1]
```diff
@@ -168,6 +168,10 @@ void RSkComponentTextInput::onHandleKey(rnsKey eventKeyType, bool *stopPropagati
     textInputMetrics.contentSize.height = paragraph_->getHeight();
     textInputEventEmitter->onFocus(textInputMetrics);
     isInEditingMode_ = true;
+    if (!caretHidden_) {
+      std::cout<<"after onFocus!!!!!!!!!"<<std::endl;
```
Comment (pprabhu-ng) [inline ReactSkia/components/RSkComponentTextInput.cpp]: if the logging message is required , please use RNS_LOG_*
Same comment for the below log messages.

**Comment by chatrapathi06** [comment]
Now handle Command API implementation is done. you need to add your changes to focus and blur method.

File: ReactSkia/components/RSkComponentTextInput.cpp [hunk 2]
```diff
@@ -241,7 +245,10 @@ void RSkComponentTextInput::processEventKey (rnsKey eventKeyType,bool* stopPropa
           }
           *stopPropagation = true;
           keyPressMetrics.eventCount = eventCount_;
-          flushLayer();
+          if (!caretHidden_) {
+            std::cout<<"after Key press left!!!!!!!!!"<<std::endl;
```
Comment (chatrapathi06) [inline ReactSkia/components/RSkComponentTextInput.cpp]: Add RNS_LOG_DEBUG, if it is required.

File: ReactSkia/components/RSkComponentTextInput.cpp [hunk 3]
```diff
@@ -252,7 +259,10 @@ void RSkComponentTextInput::processEventKey (rnsKey eventKeyType,bool* stopPropa
           }
           *stopPropagation = true;
           keyPressMetrics.eventCount = eventCount_;
-          flushLayer();
+          if (!caretHidden_) {
+            std::cout<<"after Key press right!!!!!!!!!"<<std::endl;
```
Comment (chatrapathi06) [inline ReactSkia/components/RSkComponentTextInput.cpp]: Add RNS_LOG_DEBUG, if it is required.

File: ReactSkia/components/RSkComponentTextInput.cpp [hunk 4]
```diff
@@ -271,6 +281,10 @@ void RSkComponentTextInput::processEventKey (rnsKey eventKeyType,bool* stopPropa
           textInputEventEmitter->onBlur(textInputMetrics);
           isInEditingMode_ = false;
           *stopPropagation = true;
+          if (!caretHidden_) {
+            std::cout<<"after Key press Enter!!!!!!!!!"<<std::endl;
```
Comment (chatrapathi06) [inline ReactSkia/components/RSkComponentTextInput.cpp]: Add RNS_LOG_DEBUG, if it is required.

File: ReactSkia/components/RSkComponentTextInput.cpp [hunk 5]
```diff
@@ -89,7 +89,7 @@ void RSkComponentTextInput::drawTextInput(SkCanvas *canvas,
   paragraph_->paint(canvas, frame.origin.x + layout.contentInsets.left, frame.origin.y + layout.contentInsets.top + yOffset);
 
   // draw Cursor
-  if (!caretHidden_) {
+  if (!caretHidden_ && isInEditingMode_) {
```
Comment (chatrapathi06) [inline ReactSkia/components/RSkComponentTextInput.cpp]: if possible, interchange the variable. if it is not in Editing mode no need to check caret Hidden right ?  
if (isInEditingMode_ && !caretHidden_ )

**Pull request closed without merging**